nest-forms-backend/src/utils/subservices/sharepoint.subservice.ts

```diff
@@ -0,0 +1,357 @@
+            value,
+            { ...form, title: formTitle },
+            jsonDataExtraDataOmitted,
+            oneToOneFields,
+          )
+
+          const addedId = await this.postDataToSharepoint(
+            value.databaseName,
+            accessToken,
+            valuesForFieldsOneToOne,
+          )
+
+          valuesForFields[
+            `${oneToOneOriginalTableFields[value.originalTableId]}Id`
+          ] = addedId
+        }),
+      )
+    }
+
+    const baseId = await this.postDataToSharepoint(
+      sharepointData.databaseName,
+      accessToken,
+      valuesForFields,
+    )
+
+    if (sharepointData.oneToMany) {
+      await Promise.all(
+        Object.entries(sharepointData.oneToMany).map(async ([key, value]) => {
+          const recordsArray = getArrayForOneToMany(
+            { ...form, jsonDataExtraDataOmitted },
+            key,
+          )
+
+          await Promise.all(
+            recordsArray.map(async (record) => {
+              const foreignFields: Record<string, any> = {}
+              foreignFields[value.originalTableId] = baseId
+
+              const oneToManyFields = await this.mapColumnsToFields(
+                [
+                  ...Object.keys(value.columnMap),
+                  ...Object.keys(foreignFields),
+                ],
+                accessToken,
+                value.databaseName,
+              )
+
+              const valuesForFieldsOneToMany = getValuesForFields(
+                value,
+                { ...form, title: formTitle },
+                record,
+                oneToManyFields,
+                foreignFields,
+              )
+
+              await this.postDataToSharepoint(
+                value.databaseName,
+                accessToken,
+                valuesForFieldsOneToMany,
+              )
+            }),
+          )
+        }),
+      )
+    }
+
+    await this.prismaService.forms.update({
+      where: {
+        id: formId,
+      },
+      data: {
+        error: FormError.NONE,
+        state: FormState.PROCESSING,
+      },
+    })
+  }
+
+  private async mapColumnsToFields(
+    columns: Array<string>,
+    accessToken: string,
+    dtbName: string,
+  ): Promise<Record<string, string>> {
+    const result: Record<string, string> = {}
+    const { SHAREPOINT_URL } = process.env
+    const url = `${escape(SHAREPOINT_URL)}/lists/getbytitle('${dtbName}')/fields`
+
+    const fields = await axios
+      .get(url, {
+        headers: {
+          Accept: 'application/json;odata=verbose',
+          'Content-Type': 'application/json;odata=verbose',
+          Authorization: `Bearer ${accessToken}`,
+        },
+      })
+      .then(
+        (
+          response: AxiosResponse<
+            { d: { results: List<Record<string, string>> } },
+            object
+          >,
+        ) => response.data.d.results,
+      )
+
+    columns.forEach((col) => {
+      const filtered = filter<Record<string, string>>(
```

Comment:
Do we need lodash filter? Also, it seems that you are looking for one instance, why not use `Array.find`? Then you don't have to depend on "unsafe" `[0]`.